### PR TITLE
Add a -Z time option which prints only passes which runs once

### PR DESCRIPTION
**Pull request by Zoxc**
This ensures `-Z time-passes` fits on my screen =P

r? @michaelwoerister

### CODE REVIEW

**Comment by michaelwoerister** [comment]
What does it do?

File: src/librustc_typeck/coherence/mod.rs [hunk 1]
```diff
@@ -133,9 +133,7 @@ fn coherent_trait<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId) {
     for &impl_id in impls {
         check_impl_overlap(tcx, impl_id);
     }
-    use rustc::util::common::time;
-    time(tcx.sess, "builtin::check_trait checking", ||
-          builtin::check_trait(tcx, def_id));
```
Comment (Zoxc) [inline src/librustc_typeck/coherence/mod.rs]: I added this for profiling purposes, but forgot to remove it.

**Comment by Zoxc** [comment]
`-Z time-extended` does what `-Z time-passes` currently does. While `-Z time-passes` is changed to only show passes which run once.

**Comment by michaelwoerister** [comment]
I'd prefer not to change the behaviour of `-Ztime-passes`. Would you be OK with making `-Ztime-passes` take an argument for the behaviour you want and letting it default to the current behaviour? So `-Ztime-passes` would do the same as now but one can do `-Ztime-passes=reduced` to get the smaller output.

**Comment by Zoxc** [comment]
I think I'd rather add a `-Z time` option, which defaults to the smaller output.

**Comment by michaelwoerister** [comment]
That's fine too. I just don't want to change the behaviour of `-Ztime-passes` unnecessarily.

**Comment by Dylan-DPC** [comment]
ping from triage @Zoxc any updates on this?

**Comment by TimNN** [comment]
Ping from triage @Zoxc: What is the status of this PR?

**Comment by Centril** [comment]
Ping from triage, @michaelwoerister thoughts?

**Comment by michaelwoerister** [comment]
Oh, I didn't see that an update was pushed here. 
@bors r+

**Pull request merged**